runtime/configs/config.go

```diff
@@ -92,15 +107,60 @@ func ParseConfig(cfgFile string) *Config {
 	return config
 }
 
-// setViperDefaults this is a hacky way to set the default values for Viper so env var overrides work.
+// adjustConfigDefaults performs the necessary adjustments on the default configuration depending on the provided custom configuration.
```

Comment:
Maybe `hydrateDefaultConfigs` is a better choice of words for this?

Not a blocker, but just thinking out loud

Reply:
Thank you for the review. I reverted changes related to config, as they had resulted in too many lines of code being modified without fully resolving the config issue.
I opened an issue instead to track the incomplete default configuration: #858